models/location.js

```diff
@@ -0,0 +1,60 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+// create Location Schema & model
+const LocationSchema = new Schema({
+  locationId: {
+    type: Schema.Types.ObjectId,
+    required: [true, "locationId is required to save Location"]
+  },
+  groupId: {
```

Comment:
UML relates location to Mark, not Group. Then the group will end up relating to Mark.